### PR TITLE
Try and fix the service-provider issue

### PR DESCRIPTION
**Pull request by JackMc**
This is what's used to generate the images on Docker Hub.

### CODE REVIEW

**Comment by deniszgonjanin** [comment]
My PR to ckan-service-provider just got merged an hour ago. I'm fine with leaving this PR as it is though, @wardi can decide

File: docker/ckan/Dockerfile [hunk 1]
```diff
@@ -10,6 +10,7 @@ RUN virtualenv $CKAN_HOME && \
     chown -R www-data:www-data /var/www && \
     git clone --depth 1 --branch $BRANCH https://github.com/ckan/ckan /project/ckan && \
     git clone --branch stable --depth 1 https://github.com/ckan/datapusher /project/datapusher && \
+    $CKAN_HOME/bin/pip install -e git+https://github.com/deniszgonjanin/ckan-service-provider#egg=ckanserviceprovider && \
```
Comment (wardi) [inline docker/ckan/Dockerfile]: anything we install here gets blown away when a user runs `install --clean` so we would need to add this hack there too :-(

**Comment by deniszgonjanin** [comment]
datapusher master's ckan-service-provider requirement got bumped to 0.0.5, which has the SQLAlchemy fix, and a new version of provider was released on pypi. Parts of this PR could be reverted to point to the official repos again.

**Comment by JackMc** [comment]
Yay! Unfortunately we run on data pusher's stable branch :-(

**Comment by deniszgonjanin** [comment]
Ah. Damn

File: docker/ckan/Dockerfile (hunk 1 above)
Comment (wardi) [inline docker/ckan/Dockerfile]: @JackMc this is the last thing holding up this PR, would you confirm that `install --clean` leaves us in a broken state? And if so, other than forcing install of this branch like we do for ckanapi is there another way we could fix this problem?

**Pull request merged**